storage/remote/otlptranslator/prometheusremotewrite/helper.go

```diff
@@ -118,7 +118,7 @@ var seps = []byte{'\xff'}
 // if logOnOverwrite is true, the overwrite is logged. Resulting label names are sanitized.
 // If settings.PromoteResourceAttributes is not empty, it's a set of resource attributes that should be promoted to labels.
 func createAttributes(resource pcommon.Resource, attributes pcommon.Map, scope scope, settings Settings,
-	ignoreAttrs []string, logOnOverwrite bool, metadata prompb.MetricMetadata, extras ...string,
+	ignoreAttrs []string, logOnOverwrite bool, metadata prompb.MetricMetadata, temporality pmetric.AggregationTemporality, hasTemporality bool, extras ...string,
```

Comment:
Instead of `hasTemporality`, can't you use `AggregationTemporalityUnspecified` to signal that temporality is unset? It would simplify the logic greatly I think, and is more idiomatic enum handling.